services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -5682,7 +5772,16 @@ public String getNativeReplicationSourceFabric(
     }
 
     if (sourceFabric == null || sourceFabric.isEmpty()) {
-      sourceFabric = multiClusterConfigs.getControllerConfig(clusterName).getNativeReplicationSourceFabric();
+      sourceFabric = getMultiClusterConfigs().getControllerConfig(clusterName).getNativeReplicationSourceFabric();
+    }
+
+    if (targetedRegions != null) {
+      Set<String> regions = RegionUtils.parseRegionsFilterList(targetedRegions);
+      if (!regions.contains(sourceFabric) && !emergencySourceRegion.isPresent()) {
```

Comment:
Do we want to warn the user for this case? Ideally user should use the originating data fabric + other colo if they want a 2nd colo. Since its a user given config now, user could provide non-native colo, which would be very inefficient.

Reply:
Native colo/region is set by this method, so when user provides a region that isn't selected by criteria above,  whatever they provide will be selected as native colo/region.

Though we may have preferred native region to choose from, it's not necessary inefficient from my understanding and it's okay to let customers to specify it. Let me know if I miss anything here.

Reply:
Let's instead fail the job if user doesn't specify the source region as a targeted region?

Reply:
One gotcha might be if the user provides an invalid region, which cannot be used to find the corresponding kakfa bootstrap url, the existing behavior of this is parent controller will use the existing kafka config ([code](https://github.com/linkedin/venice/blob/d689f173e461b51e531e41bcfb685057637995cb/services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java#L2358)), and none of existing child regions will create a new version, eventually the push job will fail due to time out.

Reply:
Discussed offline with @majisourav99 . The reason why it can be inefficient is HDFS/Samza reprocessing job may be always from one region A (in other words, largely depends on how offline job is setup), and if the targeted regions are region B and C, although it should work with no issue, additional data hop from region A to region B or region C can add up the overall push time. It's preferred that users always include the region A in this config and specify it as the source fabric, such that it will be always get selected to reduce the extra hop

Reply:
I will try to add some logic in VPJ.
When controller query store details in VPJ, we also check the store's config of native region setup to see if the provided targeted regions contain the region or not. If not, we will give a warning in VPJ so people can see it in the logs. Maybe we can expose one controller API for it but I don’t think it is worth the effort just for a warning msg.

Reply:
We decided to introduce 2 configs. One config is boolean (easy mode) so store config will automatically be used as the targeted region, so customers don't have to do anything. The advanced mode is we can take a list of strings as regions but it can impact the performance if customers specifies a region other than where data comes from.